pkg/repl/repl.go

```diff
@@ -0,0 +1,71 @@
+// Package repl is a simple read-eval-print loop.  It calls the Consumer
+// to do all the eval work.
+package repl
+
+import (
+	"io"
+
+	"github.com/peterh/liner"
+)
+
+type Consumer interface {
+	Consume(line string) bool
+	Prompt() string
+}
+
+type REPL struct {
+	consumer Consumer
+}
+
+func NewREPL(c Consumer) *REPL {
+	return &REPL{consumer: c}
+}
+
+// Run executes the REPL.
+func (r *REPL) Run() error {
```

Comment:
Since the only thing you can do with a REPL is call its Run method, you could drop the type and NewREPL and just have a Run function:
```suggestion
func Run(c Consumer) error {
```

Reply:
Excellent call.